### PR TITLE
doc(headless): update origin level 2 for listing

### PR DESCRIPTION
**Pull request by rtherien**
DOC Jira: https://coveord.atlassian.net/browse/DOC-14196

Details: 

Slack conversation: https://coveo.slack.com/archives/C03BM5PFYBX/p1706545628261539

Discuss: https://discuss.coveo.com/t/need-to-confirm-if-overriding-tab-originlevel2-with-category-name-is-still-the-best-practice-for-listing-pages/9450

### CODE REVIEW

File: packages/headless/src/app/engine-configuration.ts [hunk 1]
```diff
@@ -110,11 +110,15 @@ export interface AnalyticsConfiguration {
    */
   originContext?: string;
   /**
-   * Origin level 2 is a usage analytics event metadata whose value should typically be the name/identifier of the tab from which the usage analytics event originates.
+   * Sets the value of the Origin Level 2 dimension on the analytic events.
    *
-   * When logging a Search usage analytics event, originLevel2 should always be set to the same value as the corresponding tab (parameter) Search API query parameter so Coveo Machine Learning models function properly, and usage analytics reports and dashboards are coherent.
+   * Origin level 2 is a usage analytics event metadata whose value should typically be the name/identifier of the tab from which the usage analytics event originates. 
    *
```
Comment (eguyon-coveo) [inline packages/headless/src/app/engine-configuration.ts:116]: To get rid of those linting warnings, make sure your branch is up-to-date with master, then run `npm i `from the root of the project to install the current version of prettier locally, and then `npm run lint:fix`. Commit and push the changes to your branch and you're done (all credits go to @fbeaudoincoveo)

File: packages/headless/src/app/engine-configuration.ts [hunk 2]
```diff
@@ -110,11 +110,15 @@ export interface AnalyticsConfiguration {
...
-   * This value is optional, and will automatically try to resolve itself from the tab search parameter.
+   * In the context of product listing, the value should match the breadcrumb of the product listing page from which the usage analytics event originates (for example, `canoes-kayaks/kayaks/sea-kayaks`).
+   *
+   * When logging a usage analytics event, originLevel2 should always be set to the same value as the corresponding `tab` (parameter) Search API query parameter so Coveo Machine Learning models function properly, and usage analytics reports and dashboards are coherent.
```
Comment (jpmarceau) [inline packages/headless/src/app/engine-configuration.ts:119]: Just so I follow: this means that even in the context of product listing, originLevel2 will match the `tab` Search API query param, which is also the breadcrumb value?

File: packages/headless/src/app/engine-configuration.ts [hunk 3]
```diff
@@ -110,11 +110,15 @@ export interface AnalyticsConfiguration {
    */
   originContext?: string;
   /**
-   * Origin level 2 is a usage analytics event metadata whose value should typically be the name/identifier of the tab from which the usage analytics event originates.
+   * Sets the value of the Origin Level 2 dimension on the analytic events.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/app/engine-configuration.ts:113]:
```suggestion
   * Sets the value of the Origin Level 2 dimension on the analytics events.
```

I see another instance of it in the originContext docs if you want to fix it as well 😄

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/app/engine-configuration.ts (hunk 2 above)
Comment (rtherien) [reply packages/headless/src/app/engine-configuration.ts:119]: @jpmarceau Yes!

Here's the payload of a listing query. You can see that the tab parameter also has the listing's breadcrumb as a value:
Screenshot 2024-02-02 at 9 40 43 AM

**Review by jpmarceau — APPROVED** [review]
👍

**Review by olamothe — APPROVED** [review]

**Pull request merged**